src/sql/workbench/api/common/extHostModelView.ts

```diff
@@ -1600,25 +1616,30 @@ class DeclarativeTableWrapper extends ComponentWrapper implements azdata.Declara
 
 			properties.data = properties.data.map((row: any[]) => row.map(cell => {
 				if (cell instanceof ComponentWrapper) {
-					// First ensure that we register the component using addItem
-					// such that it gets added to the ModelStore. We don't want to
-					// make the table component an actual container since that exposes
-					// a lot of functionality we don't need.
-					componentsToAdd.push(cell);
+					if (!this.itemConfigs.find(item => item.component.id === cell.id)) {
```

Comment:
Main fix 3

Note that technically this is redundant since when we call addItems the logic I added above will filter out already-added items. But doing this avoids the warnings so I opted to be cleaner here even if there's a slight performance hit.